src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetHelper.cs

```diff
@@ -230,5 +234,27 @@ private static string GetBinLogOption(string projectName, string command, string
         return $"/bl:{Path.Combine(LogsDirectory, $"{fileName}.binlog")}";
     }
 
+    private static bool DetermineIsMonoRuntime()
```

Comment:
I'm open to suggestions on improving this method.

My first go was to check if the running .NET was using a Mono runtime, but that didn't work because my test host was actually CoreCLR (and .NET under test was mono).

Instead we need to peek in the .NET folder and find something to key off.
I'm currently using the presence of the mono-gc.h file.

Reply:
another option would be to somehow flow the value of SourceBuildUseMonoRuntime (e.g. via a define) so we can use that to decide, but not sure that is really better.

Reply:
IMO it doesn't matter to much on the approach.  I hope to see the runtime issue resolved so there is no need for this method.